src/client.rs

```diff
@@ -461,6 +503,11 @@ impl Client {
                 reqwest::StatusCode::UNAUTHORIZED => {
                     Err(OciDistributionError::UnauthorizedError { url })
                 }
+                s if s.is_success() => Err(OciDistributionError::SpecViolationError(format!(
```

Comment:
Note to our future selves: looks like our status error handling code needs to be put into a single function somewhere

Reply:
Good point. Done that too